src/async.cpp

```diff
@@ -138,8 +142,27 @@ static int fuse_read(const char *path,
 	return size;
 }
 
+static int fuse_readlink(const char *path, char *buf, size_t size) {
+	auto *ctx = (struct _fuse_context *)fuse_get_context()->private_data;
+	if(ctx == NULL) {
+		return -EINVAL;
+	}
+
+	std::unique_lock l(ctx->m_mu);
+	ctx->done = false;
+	memset(buf, 0, size);
+	ctx->buf = buf;
+	int pid = 0;
+	char entry[32];
+	sscanf(path, "/%d/%s", &pid, entry);
+	generate_async_event(ctx->async_event_handler, ASYNC_EVENT_ENTRY_NAME, pid, entry);
+	ctx->m_cv.wait(l, [ctx] { return ctx->done; });
+	return 0;
```

Comment:
why do we return `0` and not `strlen(buf) + 1;` like in `fuse_read`?

Reply:
@Andreagit97 the return value is 0 for success https://github.com/libfuse/libfuse/blob/7a92727d97c10290b3501d86a194738973edb61d/include/fuse.h#L338-L346

Reply:
error handling could be improved though 😄

Reply:
ok thank you